src/Controller/UI/GRPMetadataExportController.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Controller\UI;
+
+use App\Entity\Dataset;
+use App\Util\FundingOrgFilter;
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface as SerializerInterface;
+
+/**
+ * GRP metadata export report generator.
+ */
+class GRPMetadataExportController extends ReportController
+{
+    /**
+     * This is a parameterless report, so all is in the default action.
+     *
+     * @return Response A Response instance.
+     */
+    #[Route('/grp/export')]
+    public function defaultAction(EntityManagerInterface $entityManager, SerializerInterface $serializer, FundingOrgFilter $fundingOrgFilter)
```

Comment:
Instead of EntityManager, is the Repository you need. In this case, DatasetRepository.

Reply:
yeah, but I can't inject that, because I need a specific one.

Reply:
Yes you can, look at other controllers.